qiskit/pulse/builder.py

```diff
@@ -623,30 +603,37 @@ def build(backend=None,
     Args:
         backend (Union[Backend, BaseBackend]): A Qiskit backend. If not supplied certain
             builder functionality will be unavailable.
-        schedule: a *mutable* pulse Schedule in which your pulse program will
+        schedule: A *mutable* pulse ``ScheduleBlock`` in which your pulse program will
             be built.
-        name: Name of pulse program to be built. Only used if `schedule`
-            is not ``None``.
+        name: Name of pulse program to be built.
         default_alignment: Default scheduling alignment for builder.
-            One of ``left``, ``right``, ``sequential`` or an alignment
-            contextmanager.
+            One of ``left``, ``right``, ``sequential`` or an alignment context.
         default_transpiler_settings: Default settings for the transpiler.
         default_circuit_scheduler_settings: Default settings for the
             circuit to pulse scheduler.
 
     Returns:
         A new builder context which has the active builder initialized.
     """
+    if isinstance(schedule, Schedule):
+        warnings.warn('Starting the builder with `Schedule` is being deprecated. '
+                      'Input `ScheduleBlock` instead.', DeprecationWarning)
```

Comment:
Again, it seems trivial to just make a schedule into a basic ScheduleBlock. Might as well accept both if we are going to accept any initial value

Reply:
The same reason as I wrote above. This internally generates new block by calling the input schedule and return the block from the builder context. This means schedule is no longer *mutable*. This is breaking change.

Reply:
going to resolve here and continue that thread